### PR TITLE
Add wol extension

### PR DESCRIPTION
**Pull request by bxb100**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/wol/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                icon={Icon.Play}
+                onAction={async () => {
+                  wake(item.mac, { port: parseInt(item.port) });
+                  await showToast({ title: "Order acknowledged", message: "For the Swarm" });
+                }}
+              />
+              <Action.CopyToClipboard title="Copy Mac Address" content={item.mac} />
+              <CreateWakeAction onCreate={handleCreate} />
+              <Action
+                title="Remove Wake Data"
```
Comment (pernielsentikaer) [inline extensions/wol/src/index.tsx:160]:
```suggestion
                title="Remove Wake Data"
                style={Action.Style.Destructive}
                shortcut={Keyboard.Shortcut.Common.Remove}
```

File: extensions/wol/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  "icon": "command-icon.png",
+  "author": "Lemon",
+  "categories": [
+    "System"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Devices",
```
Comment (pernielsentikaer) [inline extensions/wol/package.json:15]:
```suggestion
      "title": "List Devices",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- The extension crashes if you're trying to wake a device that has some malformed data

wol-1

- I entered some random data, it says online?

wol 2024-03-06 at 09 07 32

- Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by bxb100** [comment]
Thanks review
> The extension crashes if you're trying to wake a device that has some malformed data

fixed, form validate aligns with the logic in `wol.ts`

> I entered some random data, it says online?

it simply use `ping` to check the IP online status, so it works as expected

> Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

em, `capture-raycast-metadata` not working on my computer, and I check [doc](https://github.com/raycast/extensions/blob/fa7f40fce6143268c0fe497f7abce98bd8ecc23e/extensions/capture-raycast-metadata/README.md) yet don't know how to use it

image

**Comment by pernielsentikaer** [comment]
Can you try to restart Raycast (and check in Activity monitor that only one instance is running 🙂)

**Comment by bxb100** [comment]
> Can you try to restart Raycast (and check in Activity monitor that only one instance is running 🙂)

same, doesn't work

**Comment by pernielsentikaer** [comment]
Do you mind running this in a terminal: `log stream --predicate "subsystem == 'com.raycast.macos'" --level debug --style compact >> ~/Desktop/ray.$(date +%Y%m%d_%H%M%S).log` and then try again - after trying just stop the log and send the output so I can see if anything pops up 🙂

Are you in our Slack community?

**Review by pernielsentikaer — APPROVED** [review]

**Pull request merged**